### PR TITLE
Release

### PR DESCRIPTION
**Pull request by jayoshih**
All of the latest changes to rice cooker

### CODE REVIEW

File: ricecooker/config.py [hunk 1]
```diff
@@ -21,26 +21,35 @@
...
 # Folder to store progress tracking information
-RESTORE_DIRECTORY = "restore/"
+RESTORE_DIRECTORY = "restore"
 
 def get_storage_path(filename):
     """ get_storage_path: returns path to storage directory for downloading content
         Args: filename (str): Name of file to store
         Returns: string path to file
     """
+    directory = os.path.join(STORAGE_DIRECTORY, filename[0], filename[1])
```
Comment (jayoshih) [inline ricecooker/config.py]: Make sure storage directory exists

File: ricecooker/config.py [hunk 2]
```diff
@@ -99,12 +120,33 @@ def create_channel_url(domain):
     """
     return CREATE_CHANNEL_URL.format(domain=domain)
 
-def open_channel_url(channel, domain, token):
+def add_nodes_url(domain):
```
Comment (jayoshih) [inline ricecooker/config.py]: Chunk process of uploading tree to avoid timeouts

File: ricecooker/config.py [hunk 3]
```diff
@@ -99,12 +120,33 @@ def create_channel_url(domain):
...
         Args:
-            invitation (str): invitation id to get editing access
             channel (str): channel id of uploaded channel
             domain (str): server where channel was created
         Returns: string url to open channel
     """
-    return OPEN_CHANNEL_URL.format(token=token, domain=domain, channel_id=channel)
+    return OPEN_CHANNEL_URL.format(domain=domain, channel_id=channel)
+
+def publish_channel_url(domain):
```
Comment (jayoshih) [inline ricecooker/config.py]: Add option to publish channel

File: ricecooker/managers.py [hunk 4]
```diff
@@ -88,121 +89,142 @@ def download_graphie(self, path, title):
             Args: path (str): path to .svg and .json files
             Returns: the combined hash of graphie files and their filenames
         """
+        # Handle if path has already been processed
```
Comment (jayoshih) [inline ricecooker/managers.py]: Change graphie write

File: ricecooker/managers.py [hunk 5]
```diff
@@ -412,15 +439,59 @@ def upload_tree(self, token):
...
-            "file_data": self.downloader._file_mapping,
         }
-        response = requests.post(config.create_channel_url(self.domain), data=json.dumps(payload))
+        response = requests.post(config.create_channel_url(self.domain), headers={"Authorization": "Token {0}".format(token)}, data=json.dumps(payload))
         response.raise_for_status()
         new_channel = json.loads(response._content.decode("utf-8"))
-        return config.open_channel_url(new_channel['new_channel'], self.domain, token)
+        return new_channel['root'], new_channel['channel_id']
+
+    def add_nodes(self, root_id, children, token):
```
Comment (jayoshih) [inline ricecooker/managers.py]: Split out uploading tree process

**Comment by jamalex** [comment]
A lot of other PR's depend on this, so I reviewed and created Trello cards for a couple things I saw. Going to merge now, to make reviewing the other PRs easier.

**Pull request merged**